cloud/amazon/route53.py

```diff
@@ -218,8 +224,9 @@ def main():
     zones = {}
     results = conn.get_all_hosted_zones()
     for r53zone in results['ListHostedZonesResponse']['HostedZones']:
-        zone_id = r53zone['Id'].replace('/hostedzone/', '')
-        zones[r53zone['Name']] = zone_id
+        if (r53zone['Config']['PrivateZone'] == 'false' and not module.params['private_zone']) or (r53zone['Config']['PrivateZone'] == 'true' and module.params['private_zone']):
+            zone_id = r53zone['Id'].replace('/hostedzone/', '')
```

Comment:
It would fit the style of ansible modules more to set a variable around line 189 that we could compare here. Something like:

``` python
     retry_interval_in     = module.params.get('retry_interval')
     if module.params.get('private_zone'):
         private_zone_in = 'true'
     else:
         private_zone_in = 'false'
[...]
         if r53zone['Config']['PrivateZone'] == private_zone_in:
[...]
```